backend/app/bedrock.py

```diff
@@ -291,21 +295,55 @@ def calculate_price(
-    if model == "claude-v2":
-        return "anthropic.claude-v2:1"
-    elif model == "claude-instant-v1":
-        return "anthropic.claude-instant-v1"
-    elif model == "claude-v3-sonnet":
-        return "anthropic.claude-3-sonnet-20240229-v1:0"
-    elif model == "claude-v3-haiku":
-        return "anthropic.claude-3-haiku-20240307-v1:0"
-    elif model == "claude-v3-opus":
-        return "anthropic.claude-3-opus-20240229-v1:0"
-    elif model == "claude-v3.5-sonnet":
-        return "anthropic.claude-3-5-sonnet-20240620-v1:0"
-    elif model == "mistral-7b-instruct":
-        return "mistral.mistral-7b-instruct-v0:2"
-    elif model == "mixtral-8x7b-instruct":
-        return "mistral.mixtral-8x7b-instruct-v0:1"
-    elif model == "mistral-large":
-        return "mistral.mistral-large-2402-v1:0"
+
+    # Base model IDs for cross-region inference
+    base_model_ids = {
+        "claude-v2": "anthropic.claude-v2:1",
+        "claude-instant-v1": "anthropic.claude-instant-v1",
+        "claude-v3-sonnet": "anthropic.claude-3-sonnet-20240229-v1:0",
+        "claude-v3-haiku": "anthropic.claude-3-haiku-20240307-v1:0",
+        "claude-v3-opus": "anthropic.claude-3-opus-20240229-v1:0",
+        "claude-v3.5-sonnet": "anthropic.claude-3-5-sonnet-20240620-v1:0",
+        "mistral-7b-instruct": "mistral.mistral-7b-instruct-v0:2",
+        "mixtral-8x7b-instruct": "mistral.mixtral-8x7b-instruct-v0:1",
+        "mistral-large": "mistral.mistral-large-2402-v1:0",
+    }
+
+    # Models that support cross-region inference
+    CROSS_REGION_INFERENCE_MODELS = {
+        "claude-v3-sonnet",
+        "claude-v3-haiku",
+        "claude-v3-opus",
+        "claude-v3.5-sonnet",
+    }
+
+    base_model_id = base_model_ids.get(model)
+    if not base_model_id:
+        raise ValueError(f"Unsupported model: {model}")
+
+    if (
+        ENABLE_BEDROCK_CROSS_REGION_INFERENCE
+        and is_region_supported_for_cross_inference(BEDROCK_REGION)
+        and model in CROSS_REGION_INFERENCE_MODELS
+    ):
+        logger.info(
+            f"Using cross-region inference for model {model} in region {BEDROCK_REGION}"
+        )
+        return base_model_id
+    else:
+        if ENABLE_BEDROCK_CROSS_REGION_INFERENCE:
+            if not is_region_supported_for_cross_inference(BEDROCK_REGION):
+                logger.warning(
+                    f"Cross-region inference is enabled, but the region {BEDROCK_REGION} is not supported. Using local model."
+                )
+            elif model not in CROSS_REGION_INFERENCE_MODELS:
+                logger.warning(
+                    f"Cross-region inference is not available for model {model}. Using local model."
+                )
+
+        # Adjust the base_model_id to use the local model ID
+        # For local models, may need to adjust the model ID format
+        region_prefix = BEDROCK_REGION
+        local_model_id = f"{region_prefix}/{base_model_id}-local"
```

Comment:
Please read the doc carefully to enable cross region inference
https://docs.aws.amazon.com/bedrock/latest/userguide/cross-region-inference-support.html

Reply:
Thank you. Made the following changes taking into account the AWS documentation:

1. Removed unnecessary `model_regions` mapping and associated conditions
2. Created `region_suffixes` dictionary that maps `BEDROCK_REGION` directly to the appropriate region suffix (`us` or `eu`)
3. Defined `cross_region_models` set containg the models that support cross-region inference.
4. If cross-region inference is enabled and the model supports it, we append the region suffix to the base model ID. Otherwise, we use the base model ID directly.
5.  Added logging statements to indicate whether we're using a cross-region model ID or the local model ID.
6. If the region does not support cross-region inference (i.e., it's not in `region_suffixes`), we log a warning and default to the base model ID.

View in 85e83810ce466b39ed6064d60f96e8be1ba37032